pippo-core/src/main/java/ro/pippo/core/route/DefaultRouter.java

```diff
@@ -416,16 +417,22 @@ private String uriFor(PatternBinding binding, Map<String, Object> parameters) {
 
         Map<String, Object> queryParameters = new HashMap<>(parameters.size());
 
+        String uri = route.getUriPattern();
+
         for (Entry<String, Object> parameterPair : parameters.entrySet()) {
             boolean foundAsPathParameter = false;
 
             StringBuffer sb = new StringBuffer();
-            String buffer = String.format(VARIABLE_PART_PATTERN_WITH_PLACEHOLDER, parameterPair.getKey());
-
-            Pattern pattern = Pattern.compile(buffer);
+            String regex = String.format(VARIABLE_PART_PATTERN_WITH_PLACEHOLDER, parameterPair.getKey());
+            Pattern pattern = Pattern.compile(regex);
             Matcher matcher = pattern.matcher(uri);
             while (matcher.find()) {
                 String pathValue = parameterPair.getValue().toString();
+                if (isResourceRoute && ResourceHandler.PATH_PARAMETER.equals(parameterPair.getKey())) {
+                    String versionedResourcePath = ((UrlResourceHandler) route.getRouteHandler()).injectVersion(pathValue);
+                    log.debug("Inject version in resource path: '{}' => '{}'", pathValue, versionedResourcePath);
```

Comment:
After we finalize a design perhaps this should be `trace`.  Pippo is a little verbose when logging request processing.